packages/components/src/components/hds/form/super-select/multiple/base.ts

```diff
@@ -3,46 +3,78 @@
+} from '../types.ts';
+
+import type { PowerSelectSignature } from 'ember-power-select/components/power-select';
+import type { Select as PowerSelect } from 'ember-power-select/components/power-select';
+import type { CalculatePositionResult } from 'ember-basic-dropdown/utils/calculate-position';
+import type { HdsFormSuperSelectHorizontalPositions } from '../types.ts';
+
+export const DEFAULT_HORIZONTAL_POSITION: string =
+  HdsFormSuperSelectHorizontalPositionValues.Left;
+export const HORIZONTAL_POSITION_MAPPING =
+  HdsFormSuperSelectHorizontalPositionToPlacementValues;
+
+export interface HdsFormSuperSelectMultipleBaseSignature {
+  Args: PowerSelectSignature['Args'] & {
+    showAfterOptions?: boolean;
+    afterOptionsContent?: string;
+    resultCountMessage?: string;
+    dropdownMaxWidth?: string;
+    matchTriggerWidth?: boolean;
+    isInvalid?: boolean;
+  };
+  Blocks: PowerSelectSignature['Blocks'];
+  Element: PowerSelectSignature['Element'];
+}
 
-const DEFAULT_HORIZONTAL_POSITION = 'bottom-start';
-const HORIZONTAL_POSITION_MAPPING = {
-  left: 'bottom-start',
-  center: 'bottom',
-  right: 'bottom-end',
-};
-
-export default class HdsSuperSelectMultipleBaseComponent extends PowerSelectComponent {
-  @tracked powerSelectAPI;
+export default class HdsFormSuperSelectMultipleBaseComponent extends Component<HdsFormSuperSelectMultipleBaseSignature> {
+  @tracked powerSelectAPI?: PowerSelect;
   @tracked showOnlySelected = false;
   @tracked showNoSelectedMessage = false;
 
-  get selectedCount() {
-    return this.selected?.length || '0';
+  get horizontalPosition(): HdsFormSuperSelectHorizontalPositions {
+    const { horizontalPosition = DEFAULT_HORIZONTAL_POSITION } = this.args;
+    return horizontalPosition as HdsFormSuperSelectHorizontalPositions;
+  }
+
+  get selectedCount(): string {
+    return this.powerSelectAPI?.selected?.length || '0';
   }
 
-  get optionsCount() {
-    return this.options?.length || '0';
+  get optionsCount(): string {
+    return this.powerSelectAPI?.resultsCount.toString() || '0';
   }
 
-  get resultCountMessage() {
+  get resultCountMessage(): string {
     return `${this.selectedCount} selected of ${this.optionsCount} total`;
   }
 
-  @action calculatePosition(trigger, content) {
+  @action calculatePosition(
+    trigger: Element,
+    content: HTMLElement
+  ): CalculatePositionResult {
     // use `hds-anchored-position` to calculate and set position
+    // @ts-expect-error: known issue with type of invocation
     anchoredPositionModifier(content, [trigger], {
-      placement: this.args.horizontalPosition
-        ? HORIZONTAL_POSITION_MAPPING[this.args.horizontalPosition]
-        : DEFAULT_HORIZONTAL_POSITION,
+      placement: HORIZONTAL_POSITION_MAPPING[this.horizontalPosition],
       offsetOptions: 4,
       enableCollisionDetection: true,
     });
     // prevent PowerSelect from setting position
-    return {};
+    return {
```

Comment:
Out of curiosity, why did the return value change?

Reply:
We previously returned an empty object, forcing the PowerSelect to fall back to the default values. Now that we moved to TypeScript we need to provide the bare minimum for the `CalculatePositionResult` object to have a type match. I provided some default values for the mandatory properties, but the computation is done by the `anchoredPositionModifier` so it doesn't make much of a difference.